lib/pages/login/login_page.dart

```diff
@@ -129,34 +193,36 @@ class LoginPage extends StatelessWidget {
               ),
             ),
             child: Text(AppLocalizations.of(context)!.loginText),
-            onPressed: () {
-              // TODO login https://github.com/luongvo/flutter-survey/issues/7
-              Navigator.of(context).pushNamed(Routes.HOME_PAGE);
-            },
+            onPressed: () => _attemptLogin(),
           ),
         ),
       ],
     );
   }
 
-  InputDecoration _formInputDecoration(BuildContext context, String hint) =>
-      InputDecoration(
-        border: OutlineInputBorder(
-          borderSide: BorderSide.none,
-          borderRadius: BorderRadius.circular(
-            Dimens.inputBorderRadius,
-          ),
-        ),
-        contentPadding: const EdgeInsets.symmetric(
-          horizontal: Dimens.inputHorizontalPadding,
-          vertical: Dimens.inputVerticalPadding,
-        ),
-        fillColor: ColorName.whiteAlpha18,
-        filled: true,
-        hintStyle: Theme.of(context)
-            .textTheme
-            .bodyText1!
-            .copyWith(color: ColorName.whiteAlpha18),
-        hintText: hint,
-      );
+  String? _emailValidator(String? value) {
+    if (value == null || !EmailValidator.validate(value)) {
+      return AppLocalizations.of(context)!.validationErrorEmailInvalid;
+    }
+    return null;
+  }
+
+  String? _passwordValidator(String? value) {
+    if (value == null || value.length < PASSWORD_LENGTH_MIN) {
+      return AppLocalizations.of(context)!.validationErrorEmailInvalid;
+    }
+    return null;
+  }
```

Comment:
Should these logic move to LoginModel? 🤔

Reply:
@markgravity actually they are ui interaction, we try to return the error back to the textfield on typing 💭 So imo, it's better to keep them here

Reply:
@markgravity check [this](https://github.com/luongvo/flutter-survey/pull/51/files#diff-33389b53b7ec6b6718dcbc20e01367094076fec4fd39356ac5afcced83972816R151) 🤔